Ban Module

veryard opened the pull request:
- phpBB 3 Ban Module

veryard committed:
Ban Module

- phpBB 3 Ban Module

euantorano commented on boards/phpbb3.php:
```diff
@@ -8,83 +8,84 @@
  */
```
I assume the large diff here is due to some automatic formatting from an IDE/editor?

veryard replied:
Yeah, recently started using PHPStorm.

euantorano commented on resources/modules/bans.php:
```diff
@@ -0,0 +1,58 @@
+ * License: http://www.mybb.com/download/merge-system/license/
+ */
+
+abstract class Converted_Module_Bans extends Converter_Module
```
This should be `Converter_Module_Bans` I believe:

```suggestion
abstract class Converter_Module_Bans extends Converter_Module
```

veryard replied:
Whoops! Had it named `Converted` in both cases, renamed for consistency.

euantorano commented on boards/phpbb3/bans.php:
```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * MyBB 1.8 Merge System
+ * Copyright 2014 MyBB Group, All Rights Reserved
```
This copyright year should be 2019:

```suggestion
 * Copyright 2019 MyBB Group, All Rights Reserved
```

veryard replied:
Updated my files to 2019.

euantorano replied:
Perfect, thanks!

euantorano reviewed (changes requested):
This looks good to me, just some minor comments that I've spotted so far.

I haven't tested this at all yet, to do so I will need to spin up a PHPBB 3 install and ban some users. I'll try and get a chance to do that this weekend.

veryard committed:
Class name fix

burner1024 commented:
Just tried this.
1) All bans end today (well, at least permanent ones, I only have those).
2) Ban reason is missing.
3) Ban source is always "MyBB Engine".
4) Several "Guest" accounts somehow creeped in? Maybe those are deleted users?

![Captura de pantalla de 2019-03-22 21-20-09](https://user-images.githubusercontent.com/1279852/54829502-24342980-4ce9-11e9-9c42-066f57d0a277.png)

I don't mean to be picky, of course, just trying to help too.

veryard commented:
> Just tried this.
> 
> 1. All bans end today (well, at least permanent ones, I only have those).
> 2. Ban reason is missing.
> 3. Ban source is always "MyBB Engine".
> 4. Several "Guest" accounts somehow creeped in? Maybe those are deleted users?
> 
> ![Captura de pantalla de 2019-03-22 21-20-09](https://user-images.githubusercontent.com/1279852/54829502-24342980-4ce9-11e9-9c42-066f57d0a277.png)
> 
> I don't mean to be picky, of course, just trying to help too.

Thanks for the feedback:
1. Will have a look, MyBB has a weird dateline field that i'll need to code from phpBB.
2. phpBB has two reasons, one that is given and one that is shown, at the moment i'm just pulling the `ban_reason` not the `ban_give_reason` (I'll make it pull the `ban_give_reason` if the initial one is empty)
3. I don't think phpBB store the user id against the ban when banned? (Would need to pull from the `phpbb_log` but these can be cleared so im not to sure.
4. I'm not sure about this, I only had 2 users and they both came through no guests.

I'll get to everything on Sunday.

Thanks,
Brad.

euantorano commented:
I'll keep this as requiring changes until I've had a chance to test, then approve it if it looks like everything works. Thanks for the contribution 😄

burner1024 commented:
2. I definitely have an account with both reasons filled in, still comes up empty in MyBB.
3. I think you're right about the logs. It would be helpful to pull the source from there, but it's not crucial.
4. Yes, it appears that deleted users can leave lingering bans, db integrity is not enforced. Can probably just skip these:
```
select count(ban_id) from phpbb_banlist where ban_userid not in (select user_id from phpbb_users);
+---------------+
| count(ban_id) |
+---------------+
|             8 |
+---------------+
```

5. Also, there are IP and email bans, ideally they should be transferred as well.

veryard committed:
Updates

- Made reason pull from give reason if initial reason is empty.
- GID defaults to 7
- Ban Time added
- No longer pulls deleted or users that do not exist.

veryard commented:
Hello,

Have made changes: 
- Reason will pull from `ban_give_reason` if `ban_reason` is empty
- Deleted users or users that do not exist in the users table are ignored.
- `bantime` is now generated based from the phpBB fields `ban_start` & `ban_end`, treats bans less than one day as a one day ban. (phpBB allows users to be banned for 30 mins, 1 hour etc etc.)



> 5. Also, there are IP and email bans, ideally they should be transferred as well.

I looked into this, looks like MyBB stores these bans in the `banfilters` table which is a bit odd considering it all could be done from one table. I'll have to extend the code to do this some where down the line.

euantorano commented:
I’d be happy with email/IP bans being added at a later date, I think having just standard band would be a big enough improvement on its own. 

> On 23 Mar 2019, at 02:17, Brad Veryard <notifications@github.com> wrote:
> 
> Hello,
> 
> Have made changes:
> 
> Reason will pull from ban_give_reason if ban_reason is empty
> Deleted users or users that do not exist in the users table are ignored.
> bantime is now generated based from the phpBB fields ban_start & ban_end, treats bans less than one day as a one day ban. (phpBB allows users to be banned for 30 mins, 1 hour etc etc.)
> Also, there are IP and email bans, ideally they should be transferred as well.
> I looked into this, looks like MyBB stores these bans in the banfilters table which is a bit odd considering it all could be done from one table. I'll have to extend the code to do this some where down the line.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.

veryard commented:
Let me know if you need me to fix/update anything.

Brad.

burner1024 commented:
What if the text contains valid tags, though? Code snippets, etc.

veryard commented:
> What if the text contains valid tags, though? Code snippets, etc.

Sorry, I don't understand what you're saying?

burner1024 commented:
err... that was for the pull for issue #223

veryard commented:
> err... that was for the pull for issue #223
I see, well phpBB uses BBCodes aswell, so they are code snippets are inside the `[code]` BBCode.
